src/components/link/link.component.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { assign } from 'lodash';
+import { Link as RouterLink } from 'react-router';
+import Icon from '../icon';
+import { validProps } from '../../utils/ether';
+import Event from '../../utils/helpers/events';
+import tagComponent from '../../utils/helpers/tags';
+import LinkStyle from './link.style';
+import './link.scss';
+
+class Link extends React.Component {
+  static safeProps = ['onClick'];
+
+  constructor() {
+    super();
+    this.onKeyDown = this.onKeyDown.bind(this);
+  }
+
+  onKeyDown(ev) {
+    if (this.props.onKeyDown) {
+      this.props.onKeyDown(ev);
+    }
+
+    // return early if there is no onClick or there is a href prop
+    // or the event is not an enter key
+    if (this.props.href || !Event.isEnterKey(ev)) {
+      return;
+    }
+
+    if (this.props.onClick) {
+      this.props.onClick(ev);
+    }
+  }
+
+  renderLinkIcon = (currentAlignment = 'left') => {
+    const hasProperAlignment = this.props.icon && (this.props.iconAlign === currentAlignment);
+
+    return hasProperAlignment ? this.icon : null;
+  }
+
+  get icon() {
+    const classes = classNames(
+      'carbon-link__icon',
+      `carbon-link__icon--align-${this.props.iconAlign}`
```

Comment:
Is there a reason we're still including class names here?

Reply:
Yes, the problem is that `Icon` component is working on that, and changes need to be create also inside the `Icon` component but this is out of the ticket scope